src/macaron/repo_finder/commit_finder.py

```diff
@@ -356,6 +358,10 @@ def _build_version_pattern(name: str, version: str) -> tuple[Pattern | None, lis
         elif count > 1:
             this_version_pattern = this_version_pattern + INFIX_3
 
+        if numeric_only:
+            # Allow for any number of preceding zeros when the part is numeric only. E.g. 000 + 1, 0 + 20
+            this_version_pattern = this_version_pattern + "0*"
```

Comment:
If we are trying to match the preceding zeros, I wonder why we append the pattern for "0s" at the end, but not at the front.

Reply:
The version part gets added after this, so this addition is a preceding zero for that.

Reply:
Oh I see. Thanks for the clarification.